server.go

```diff
@@ -982,9 +984,32 @@ func (s *server) sendPeerMessages(
 		defer wg.Done()
 	}
 
+	// We queue each message, creating a slice of error channels that
+	// can be inspected after every message is successfully added to
+	// the queue.
+	var errChans []chan error
 	for _, msg := range msgs {
-		targetPeer.queueMsg(msg, nil)
+		errChan := make(chan error, 1)
+		if err := targetPeer.queueMsg(msg, errChan); err != nil {
+			return err
+		}
+		errChans = append(errChans, errChan)
 	}
+
+	// Return an error if any of the messages failed being sent.
+	for _, errChan := range errChans {
+		select {
+		case err := <-errChan:
```

Comment:
As was pointed out before, we don't _always_ want this method to block until completion. Instead, it can take the  error channel, and only block if it's nil.